GitUI/CommandsDialogs/BrowseDialog/FormGitLog.cs

```diff
@@ -135,12 +133,22 @@ private void SubscribeToEvents()
 
         private void OnCommandsLogChanged(object sender, EventArgs e)
         {
-            _syncContext.Post(_ => RefreshLogItems(), null);
+            ThreadHelper.JoinableTaskFactory.RunAsync(
```

Comment:
> When should we apply FileAndForget. I think it should be applied here, otherwise exceptions will be swallowed.

It should be applied to all tasks which are not joined somewhere. I've proposed analyzers to aid in this effort but they are not yet implemented.

Reply:
> I remember you proposed a solution with CancelationTokenSource protecting from operating on disposed form. Is it possible to you showing it in this PR or in a separate one?

Implemented the extension method in 364145592aa2b771b8c81f96b1a76c2862846d51 and updated most code locations to use it in 66d13c4cff34eb1ff50aa3287a6d68ded15af963. There are a few remaining cases that are not handled:

1. Passing a secondary `CancellationToken` (needs to create a linked token source for the context switch and make sure it's cleaned up after)
2. Cases where it's not clear what the control is
3. Cases where it's not clear that the following code is simply updating the UI thread

Reply:
Thank you!